test-app/tests/unit/models/sprite-snapshot-tree-test.ts

```diff
@@ -0,0 +1,47 @@
+import { module, test } from 'qunit';
+
+/**
+ * spriteTree: SpriteTree,
+ * contexts: Set<AnimationContext>,
+ * freshlyAdded: Set<SpriteModifier>,
+ * freshlyRemoved: Set<SpriteModifier>
+ */
+module('Unit | Util | SpriteSnapshotTree', function () {
+  test('it assigns correct sprites to each context', async function (assert) {
+    /**
+     * Correctness here means:
+     * - Kept, Removed, and Inserted sprites are all found under the context we expect them to be under
+     */
+    // create a bunch of contexts that are rendering (lift)
+    // create a bunch of modifiers that are added or removed (lift)
+    // put modifiers in freshlyAdded and freshlyRemoved
+    // construct a sprite tree ala filter-to-context-test
+    // let spriteSnapshotTree = new SpriteSnapshotTree(...inputs)
+    // assert spriteSnapshotTree.contextToNode(context) matches expected inputs for each context
+  });
+
+  test('it correctly identifies natural kept sprites', async function (assert) {
+    // create a bunch of contexts that are rendering (lift)
+    // make some of the contexts last and current bounds different
+    // create a bunch of sprite modifiers with last and current bounds changed
+    // construct a sprite tree ala filter-to-context-test
```

Comment:
Probably also worth adding some inserted sprites to make sure we don't introduce a bug where a sprite is both kept and inserted at the same time.